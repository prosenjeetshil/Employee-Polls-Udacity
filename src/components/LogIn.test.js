import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import Login from "./Login";

describe("Login component", () => {
  let store;
  let wrapper;
  let mockStore;
  let users = [
    {
      id: "1",
      name: "user1",
    },
    {
      id: "2",
      name: "user2",
    },
  ];

  beforeEach(() => {
    mockStore = configureStore();
    store = mockStore({
      users,
    });
    wrapper = mount(
      <Provider store={store}>
        <Login />
      </Provider>
    );
  });

  it("renders the Login component", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("displays a dropdown list of users", () => {
    expect(wrapper.find("Select").length).toBe(1);
    expect(wrapper.find("Option").length).toBe(3); // including the default "Select a user" option
  });

  it("updates the selected user when the dropdown value changes", () => {
    wrapper
      .find("Select")
      .simulate("change", { target: { value: users[1].id } });
    expect(wrapper.find("Select").props().value).toBe(users[1].id);
  });

  it("disables the login button when no user is selected", () => {
    expect(wrapper.find("Button").props().disabled).toBe(true);
  });

  it("enables the login button when a user is selected", () => {
    wrapper
      .find("Select")
      .simulate("change", { target: { value: users[0].id } });
    expect(wrapper.find("Button").props().disabled).toBe(false);
  });

  it("dispatches the setAuthedUser action when the login button is clicked", () => {
    wrapper
      .find("Select")
      .simulate("change", { target: { value: users[0].id } });
    wrapper.find("Button").simulate("click");
    expect(store.getActions()).toEqual([
      {
        type: "SET_AUTHED_USER",
        id: users[0].id,
      },
    ]);
  });
});
