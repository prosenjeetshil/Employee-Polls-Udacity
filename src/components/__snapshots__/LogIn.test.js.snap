// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="has-text-centered"
      >
        <div
          class="columns is-centered"
        >
          <div
            class="column is-two-fifths"
          >
            <h2
              class="is-size-2 my-5"
            >
              Login
            </h2>
            <p>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-user-lock fa-10x my-5"
                data-icon="user-lock"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M592 288H576V212.7c0-41.84-30.03-80.04-71.66-84.27C456.5 123.6 416 161.1 416 208V288h-16C373.6 288 352 309.6 352 336v128c0 26.4 21.6 48 48 48h192c26.4 0 48-21.6 48-48v-128C640 309.6 618.4 288 592 288zM496 432c-17.62 0-32-14.38-32-32s14.38-32 32-32s32 14.38 32 32S513.6 432 496 432zM528 288h-64V208c0-17.62 14.38-32 32-32s32 14.38 32 32V288zM224 256c70.7 0 128-57.31 128-128S294.7 0 224 0C153.3 0 96 57.31 96 128S153.3 256 224 256zM320 336c0-8.672 1.738-16.87 4.303-24.7C308.6 306.6 291.9 304 274.7 304H173.3C77.61 304 0 381.7 0 477.4C0 496.5 15.52 512 34.66 512h301.7C326.3 498.6 320 482.1 320 464V336z"
                  fill="currentColor"
                />
              </svg>
            </p>
            <form
              class="my-5"
            >
              <div
                class="select"
              >
                <select
                  id="login-as"
                  name="user"
                  role="combobox"
                >
                  <option
                    selected=""
                    value=""
                  >
                    Select a user
                  </option>
                  <option
                    value="mtsamis"
                  >
                    Mike Tsamis
                  </option>
                  <option
                    value="zoshikanlu"
                  >
                    Zenobia Oshikanlu
                  </option>
                </select>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="has-text-centered"
    >
      <div
        class="columns is-centered"
      >
        <div
          class="column is-two-fifths"
        >
          <h2
            class="is-size-2 my-5"
          >
            Login
          </h2>
          <p>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-user-lock fa-10x my-5"
              data-icon="user-lock"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M592 288H576V212.7c0-41.84-30.03-80.04-71.66-84.27C456.5 123.6 416 161.1 416 208V288h-16C373.6 288 352 309.6 352 336v128c0 26.4 21.6 48 48 48h192c26.4 0 48-21.6 48-48v-128C640 309.6 618.4 288 592 288zM496 432c-17.62 0-32-14.38-32-32s14.38-32 32-32s32 14.38 32 32S513.6 432 496 432zM528 288h-64V208c0-17.62 14.38-32 32-32s32 14.38 32 32V288zM224 256c70.7 0 128-57.31 128-128S294.7 0 224 0C153.3 0 96 57.31 96 128S153.3 256 224 256zM320 336c0-8.672 1.738-16.87 4.303-24.7C308.6 306.6 291.9 304 274.7 304H173.3C77.61 304 0 381.7 0 477.4C0 496.5 15.52 512 34.66 512h301.7C326.3 498.6 320 482.1 320 464V336z"
                fill="currentColor"
              />
            </svg>
          </p>
          <form
            class="my-5"
          >
            <div
              class="select"
            >
              <select
                id="login-as"
                name="user"
                role="combobox"
              >
                <option
                  selected=""
                  value=""
                >
                  Select a user
                </option>
                <option
                  value="mtsamis"
                >
                  Mike Tsamis
                </option>
                <option
                  value="zoshikanlu"
                >
                  Zenobia Oshikanlu
                </option>
              </select>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
